### PR TITLE
Tca 32 implement unit test for login happy path

### PR DESCRIPTION
**Pull request by pscacchi**
The unit test were implemented.
The class tested were:
- AuthWebDialogViewModel
- AuthorizationRepository
- AuthorizationUseCase
- ConsumableAuthUseCase
- UserAccessToken

### CODE REVIEW

File: app/src/test/java/ar/scacchipa/twittercloneapp/viewmodel/SplashViewModelTest.kt [hunk 1]
```diff
@@ -33,20 +30,4 @@ class SplashTimerMock: SplashTimerUseCase() {
     override suspend fun spendSplash(): Boolean {
         return true
     }
-}
-
```
Comment (sebastianMullerEndava) [inline app/src/test/java/ar/scacchipa/twittercloneapp/viewmodel/SplashViewModelTest.kt:37]: We would investigate a little bit more on this.

File: app/src/test/java/ar/scacchipa/twittercloneapp/domain/ConsumableAuthUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package ar.scacchipa.twittercloneapp.domain
+
+import org.junit.Assert
+import org.junit.Test
+
+class ConsumableAuthUseCaseTest {
+    @Test
+    fun consumableAuthUseCaseShouldReturnArl() {
```
Comment (sebastianMullerEndava) [inline app/src/test/java/ar/scacchipa/twittercloneapp/domain/ConsumableAuthUseCaseTest.kt:8]: error type in name of fun

File: app/src/test/java/ar/scacchipa/twittercloneapp/repository/AuthorizationRepositoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import okhttp3.ResponseBody
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import retrofit2.Response
+
+@ExperimentalCoroutinesApi
+class AuthorizationRepositoryTest {
+
+    private var authorizationRepository: IAuthorizationRepository? = null
```
Comment (sebastianMullerEndava) [inline app/src/test/java/ar/scacchipa/twittercloneapp/repository/AuthorizationRepositoryTest.kt:17]: should not be nullable in my opinion. Also please name the variable under testing with the name "subject"

File: app/src/test/java/ar/scacchipa/twittercloneapp/repository/AuthorizationRepositoryTest.kt [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    @Before
+    fun setup() {
+        authorizationRepository = AuthorizationRepository(MockAuthDataSource())
+    }
+
+    @Test
+    fun shouldReturnACorrectUserAccessToken() = runTest {
```
Comment (sebastianMullerEndava) [inline app/src/test/java/ar/scacchipa/twittercloneapp/repository/AuthorizationRepositoryTest.kt:25]: maybe add the why before the should on the name of the test would make it eassier to understand what is actually testing

File: app/src/test/java/ar/scacchipa/twittercloneapp/domain/ConsumableAuthUseCaseTest.kt (hunk 2 above)
Comment (pscacchi) [reply app/src/test/java/ar/scacchipa/twittercloneapp/domain/ConsumableAuthUseCaseTest.kt:8]: The new name is "consumableAuthUseCaseShouldReturnUrl()"

File: app/src/test/java/ar/scacchipa/twittercloneapp/repository/AuthorizationRepositoryTest.kt (hunk 3 above)
Comment (pscacchi) [reply app/src/test/java/ar/scacchipa/twittercloneapp/repository/AuthorizationRepositoryTest.kt:17]: Both were fixed.

File: app/src/test/java/ar/scacchipa/twittercloneapp/viewmodel/AuthWebDialogViewModelTest.kt [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+    @get:Rule
+    var mainCoroutineTestRule = MainCoroutineTestRule()
+
+    @get:Rule
+    val rule = InstantTaskExecutorRule()
+
+    @Before
+    fun setup() {
+        authWebDialogViewModel = AuthWebDialogViewModel(
```
[the remark this answers is not in the document]
Comment (pscacchi) [reply app/src/test/java/ar/scacchipa/twittercloneapp/viewmodel/AuthWebDialogViewModelTest.kt:28]: New the name of this function became "repoShouldReturnASuccessUserAccessToken()"
and the name of the next function became "repoShouldReturnADeniedUserAccessToken()"

File: app/src/test/java/ar/scacchipa/twittercloneapp/repository/AuthorizationRepositoryTest.kt (hunk 4 above)
Comment (pscacchi) [reply app/src/test/java/ar/scacchipa/twittercloneapp/repository/AuthorizationRepositoryTest.kt:25]: New the name of this function became "repoShouldReturnASuccessUserAccessToken()"
and the name of the next function became "repoShouldReturnADeniedUserAccessToken()"

File: app/src/test/java/ar/scacchipa/twittercloneapp/viewmodel/AuthWebDialogViewModelTest.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (pscacchi) [reply app/src/test/java/ar/scacchipa/twittercloneapp/viewmodel/AuthWebDialogViewModelTest.kt:28]: It was fixed.

**Review by sebastianMullerEndava — APPROVED** [review]

**Review by Diana-Endava — APPROVED** [review]

**Pull request merged**